src/main/scala/ch/epfl/bluebrain/nexus/kg/search/QueryResultEncoder.scala

```diff
@@ -48,22 +52,35 @@ object QueryResultEncoder {
       .as[Json](searchCtx deepMerge extraCtx)
       .map(_.removeKeys("@id").replaceContext(searchCtxUri).addContext(resourceCtxUri))
 
-  implicit def qrsEncoderJson: Encoder[QueryResults[Json]] = {
+  implicit def qrsEncoderJson(implicit searchUri: Uri): Encoder[QueryResults[Json]] = {
     implicit def qrEncoderJson: Encoder[QueryResult[Json]] = Encoder.instance {
       case UnscoredQueryResult(v, _) => v.removeKeys(nxv.originalSource.prefix)
       case ScoredQueryResult(score, v, _) =>
         v.removeKeys(nxv.originalSource.prefix) deepMerge Json.obj(nxv.score.prefix -> Json.fromFloatOrNull(score))
     }
-    def json(total: Long, list: List[QueryResult[Json]]): Json =
-      Json
+    def json(total: Long, list: List[QueryResult[Json]], sort: Option[Seq[Json]]): Json = {
+      val results = Json
         .obj(nxv.total.prefix -> Json.fromLong(total), nxv.results.prefix -> Json.arr(list.map(qrEncoderJson(_)): _*))
         .addContext(searchCtxUri)
         .addContext(resourceCtxUri)
+      sort match {
+        case Some(s) =>
+          val nextQuery =
+            Query(
+              searchUri
+                .query()
+                .toMap + (QueryDirectives.searchAfter -> s.asJson.noSpaces) - QueryDirectives.from)
+
+          results deepMerge Json.obj(nxv.next.prefix -> Json.fromString(searchUri.withQuery(nextQuery).toString()))
+        case None => results
+      }
+    }
     Encoder.instance {
       case UnscoredQueryResults(total, list) =>
-        json(total, list)
+        json(total, list, list.lastOption.flatMap(_.sort))
```

Comment:
That seems hacky. Maybe we should have a `sort` also in the `QueryResults`?
At the same time, that sort is `Option[Seq[Json]]` is weird to me. Shouldn't Sort be [SortList](https://github.com/BlueBrain/nexus-commons/blob/master/modules/core/src/main/scala/ch/epfl/bluebrain/nexus/commons/search/SortList.scala)??